src/dmd/expressionsem.d

```diff
@@ -2376,6 +2400,8 @@ private Module loadStdMath()
     return impStdMath.mod;
 }
 
+bool shared_allowed = false;
```

Comment:
Don't use globals

Reply:
there's no way around that would not involve passing a state pointer to every name-resolving function in dmd

Reply:
I can put it in a struct and thereby making the global access easier to spot.